File: backend/storage/src/main/java/sk/kasv/infinity/model/Course.java
package sk.kasv.infinity.model;

public class Course {

  private Long id;
  private String name;
  private String description;
  private Integer price;

  public Course(Long id, String name, String description, Integer price) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.price = price;
  }

  public Course() {

  }

  public Long getId() {
    return id;
  }

  public String getName() {
    return name;
  }

  public String getDescription() {
    return description;
  }
  public Integer getPrice() {
    return price;
  }

  public void setId(Long id) {
    this.id = id;
  }

  public void setName(String name) {
    this.name = name;
  }

  public void setDescription(String description) {
    this.description = description;
  }

  public void setPrice(Integer price) {
    this.price = price;
  }

  @Override
  public String toString() {
    final StringBuilder sb = new StringBuilder("Course{");
    sb.append("id=").append(id);
    sb.append(", name='").append(name).append('\'');
    sb.append(", description='").append(description).append('\'');
    sb.append(", price='").append(price).append('\'');
    sb.append('}');
    return sb.toString();
  }
}

File: src/entities/course.ts
export class Course {
    public static clone(c: Course):Course{
        return new Course(c.id, c.name, c.description, c.price);
    }

    constructor(
        public id: number,
        public name: string,
        public description: string,
        public price: number

    ){}

    public toString():string {
        return `id: ${this.id}, name: ${this.name}, description: ${this.description}, price: ${this.price}`;

    }
}